pkg/query/storeset.go

```diff
@@ -312,6 +329,21 @@ func (s *storeRef) Update(labelSets []labels.Labels, minTime int64, maxTime int6
 	s.exemplar = exemplar
 }
 
+func (s *storeRef) UpdateWithStore(labelSets []labels.Labels, minTime int64, maxTime int64, storeType component.StoreAPI, store storepb.StoreClient, rule rulespb.RulesClient, target targetspb.TargetsClient, metadata metadatapb.MetadataClient, exemplar exemplarspb.ExemplarsClient) {
```

Comment:
Nit: maybe this could be folded into `Update` and then we could simply add:

```go
if store != nil {
  s.StoreClient = store
}
```

This would save a bit of code duplication.

Reply:
is it store at this point? sounds really like just `endpointRef`.

Also the amount of arguments sounds like code smell. I wonder if we hide those behind better abstractions